### PR TITLE
AP_HAL_ChibiOS: Add hwdef for ARK_CANNODE

### PR DESCRIPTION
**Pull request by TunaLobster**
~~I have not tested this on hardware yet (waiting for an stlink to show up in the mail). I don't want to loose the nodes completely without an easy way to get them back. Should be a very capable node! I had a few questions that I wasn't sure how to handle.~~

EDIT: fixed and working now @andyp1per 

Schematic and repo for the board is here: https://github.com/ARK-Electronics/ARK_CANNODE/blob/main/Hardware/Rev%201/ARK%20CANNODE%20Rev%201%20Schematic.PDF

### CODE REVIEW

**Comment by khancyr** [comment]
@AlexKlimaj can we have a double view on this ? thanks

**Comment by TunaLobster** [comment]
Looking good now. I think the part that was blocking me was my novice level with the stlink. The node boots and can set parameters on the bus now.

File: libraries/AP_HAL_ChibiOS/hwdef/ARK_CANNODE/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+define HAL_PERIPH_ENABLE_GPS
+define HAL_PERIPH_ENABLE_MAG
+define HAL_PERIPH_ENABLE_BARO
+define HAL_PERIPH_ENABLE_RC_OUT
+define HAL_PERIPH_ENABLE_NOTIFY
+
+# GPS on first UART
+define HAL_PERIPH_GPS_PORT_DEFAULT 0
+
+define CAN_APP_NODE_NAME "org.ardupilot.ARK_CANNODE"
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/ARK_CANNODE/hwdef.dat:127]: this line is not needed, it is the default

File: libraries/AP_HAL_ChibiOS/hwdef/ARK_CANNODE/hwdef.dat [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+
+# I2C port
+PB6 I2C1_SCL I2C1
+PB9 I2C1_SDA I2C1
+
+# SPI device table
+SPIDEV icm42688 SPI1 DEVID1 IMU1_CS MODE3  2*MHZ  8*MHZ
+IMU Invensensev3 SPI:icm42688 ROTATION_ROLL_180
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/ARK_CANNODE/hwdef.dat:111]: maybe comment out IMU as we can't use it yet in AP_Periph?

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by tridge** [comment]
a README.md with description and some photos with pinout would be nice

File: libraries/AP_HAL_ChibiOS/hwdef/ARK_CANNODE/hwdef.dat (hunk 2 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/ARK_CANNODE/hwdef.dat]: Done

File: libraries/AP_HAL_ChibiOS/hwdef/ARK_CANNODE/hwdef.dat (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/ARK_CANNODE/hwdef.dat:124]: Fixed

**Comment by andyp1per** [comment]
I've added a couple of photos, but none of the periph devices have a readme and certainly none of the ARK devices

**Comment by AlexKlimaj** [comment]
Looks good to me. Could add a Readme with the pinout but they are just standard Pixhawk JST-GH.

**Review by tridge — APPROVED** [review]

**Pull request merged**